src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/Ssse3Helper.cs

```diff
@@ -0,0 +1,103 @@
+            //                                     high-nibble
+            // low-nibble  0   1   2   3   4   5   6   7   8   9   A   B   C   D   E   F
+            //         0   1   1   0   0   0   0   1   0   1   1   1   1   1   1   1   1
+            //         1   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         2   1   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         3   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         4   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         5   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         6   1   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         7   1   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         8   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         9   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         A   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         B   1   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         C   1   1   0   1   0   1   0   0   1   1   1   1   1   1   1   1
+            //         D   1   1   0   0   0   0   0   0   1   1   1   1   1   1   1   1
+            //         E   1   1   0   1   0   0   0   0   1   1   1   1   1   1   1   1
+            //         F   1   1   0   0   0   0   0   1   1   1   1   1   1   1   1   1
+            //
+            // where 1 denotes the neeed for escaping, while 0 means no escaping needed.
+            // For high-nibbles in the range 8..F every input needs to be escaped, so we
+            // can omit them in the bit-mask, thus only high-nibbles in the range 0..7 need
+            // to be considered, hence the entries in the bit-mask can be of type byte.
+            //
+            // In the Bitmask (see above) for each row (= low-nibble) a bit-mask for the
+            // high-nibbles (= columns) is created.
+
+            Debug.Assert(Ssse3.IsSupported);
+
+            Vector128<sbyte> highNibbles = Sse2.And(Sse2.ShiftRightLogical(sourceValue.AsInt32(), 4).AsSByte(), s_nibbleMaskSByte);
+            Vector128<sbyte> lowNibbles = Sse2.And(sourceValue, s_nibbleMaskSByte);
+
+            Vector128<sbyte> bitMask = Ssse3.Shuffle(s_bitMask, lowNibbles);
+            Vector128<sbyte> bitPositions = Ssse3.Shuffle(s_bitPosLookup, highNibbles);
+
+            Vector128<sbyte> mask = Sse2.And(bitPositions, bitMask);
+
+            Vector128<sbyte> ne0Gt = Sse2.CompareGreaterThan(mask, s_nullMaskSByte);
+            Vector128<sbyte> ne0Lt = Sse2.CompareLessThan(mask, s_nullMaskSByte);
+            mask = Sse2.Or(ne0Gt, ne0Lt);
+
+            return mask;
+        }
+
+        private static readonly Vector128<sbyte> s_nibbleMaskSByte = Vector128.Create((sbyte)0xF);
```

Comment:
Is there a better name for the mask `0xF`?

Reply:
I have been using the exact unicode character name, but maybe a context/usage specific name would be better:
https://www.fileformat.info/info/unicode/char/000f/index.htm